notion-core/src/main/scala/com/ambiata/notion/core/S3OutputStream.scala

```diff
@@ -11,16 +10,14 @@ import java.io._
 import scalaz._, Scalaz._, effect.IO
 
 object S3OutputStream {
-  def stream(address: S3Address, client: AmazonS3Client): RIO[OutputStream] = {
-    val tmpPath = uniqueFilePath
-    RIO.safe[OutputStream]({
-      val f = new BufferedOutputStream(new FileOutputStream(tmpPath.path))
+  def stream(address: S3Address, client: AmazonS3Client): RIO[OutputStream] = for {
+    t <- LocalTemporary.random.file
```

Comment:
Not sure about this one, given that there is a call to unsafe performIO and this is delaying when the file would be deleted, I am not sure this is really the same code, maybe it is ok, I would just want to be sure nothing would be leaked if a long running process was calling this a lot.